evolver/logutils.py

```diff
@@ -6,24 +7,63 @@
 logging.addLevelName(EVENT, "EVENT")
 
 
+class LogInfo:
+    """Helper class to provide "extra" to log records integrated with history.
+
+    This class acts similar to a dict, but with all fields within a dict under a
+    specific key and ensures the fields are json-serializable. Used with the
+    history integration, fields provided will be available in the stored record.
+    This is useful because all fields within extra are placed at the top level
+    of a logrecord dict, so otherwise there is no way to distinguish them.
```

Comment:
"data in the LogInfo class is "keyed" by an `EXTRA_KEY` that is stripped when the log event is read. This has the effect of normalizing LogInfo data, so that when it is read it appears at the same level as the Log entry's top-level properties. 

For example: self.logger.log(EVENT, "event from HW", extra=LogInfo(vial=1, event_class="something")) will produce a serialized log entry: {"level": "EVENT", "message": "event from HW", "event_class": "something", "vial": 1}

Reply:
I made update based on your text since it prob read strange (even though I wrote it, not AI 😄 )

Reply:
yeah, no ai from me either.